packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,283 @@
+		return [
+			Bar,
+			Button,
+			Dialog,
+		];
+	}
+
+	static get template() {
+		return ViewSettingsDialogTemplate;
+	}
+
+	static get styles() {
+		return viewSettingsDialogCSS;
+	}
+
+	static async onDefine() {
+		await fetchI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	get _dialogTitle() {
+		return this.i18nBundle.getText(VSD_DIALOG_TITLE_SORT);
+	}
+
+	get _okButtonLabel() {
+		return this.i18nBundle.getText(VSD_SUBMIT_BUTTON);
+	}
+
+	get _cancelButtonLabel() {
+		return this.i18nBundle.getText(VSD_CANCEL_BUTTON);
+	}
+
+	get _resetButtonLabel() {
+		return this.i18nBundle.getText(VSD_RESET_BUTTON);
+	}
+
+	get _ascendingLabel() {
+		return this.i18nBundle.getText(VSD_ORDER_ASCENDING);
+	}
+
+	get _descendingLabel() {
+		return this.i18nBundle.getText(VSD_ORDER_DESCENDING);
+	}
+
+	get _sortOrderLabel() {
+		return this.i18nBundle.getText(VSD_SORT_ORDER);
+	}
+
+	get _sortByLabel() {
+		return this.i18nBundle.getText(VSD_SORT_BY);
+	}
+
+	get _isPhone() {
+		return isPhone();
+	}
+
+	get _sortAscending() {
+		return !this.sortDescending;
+	}
+
+	get _sortDescending() {
+		return this.sortDescending;
+	}
+
+	get _settings() {
+		// return the current settings
+		const settings = {},
+			  sortOrderSelected = this._sortOrder.getSelectedItems(),
+			  sortBySelected = this._sortBy.getSelectedItems();
+
+		settings.sortOrder = sortOrderSelected.length ? sortOrderSelected[0].innerText : "";
+		settings.sortBy = sortBySelected.length ? sortBySelected[0].innerText : "";
+		return settings;
+	}
+
+	open() {
+		if (!this._dialog) {
+			this._dialog = this.shadowRoot.querySelector("[ui5-dialog]");
+			this._sortOrder = this.shadowRoot.querySelector("[ui5-list][sort-order]");
+			this._sortBy = this.shadowRoot.querySelector("[ui5-list][sort-by]");
+			this._resetButton = this.shadowRoot.querySelector("[ui5-button][reset-button]");
+			this._recentlyFocused = this._sortOrder;
+			this._dialog.ariaLabel = this._dialogTitle;
+			this._initialSettings = this._settings;
+			this._currentSettings = this._initialSettings;
+		} else {
+			this._restoreSettings(this._currentSettings);
+		}
+		this._resetButtonDisabled();
+		this._dialog.open();
+	}
+
+	close() {
+		this._dialog && this._dialog.close();
+	}
+
+	_focusLastControl() {
+		const recentlyFocusedSelectedItems = this._recentlyFocused.getSelectedItems(),
+			  recentlyFocusedItems = this._recentlyFocused.items;
+
+		if (recentlyFocusedSelectedItems.length) {
+			recentlyFocusedSelectedItems[0].focus();
+		} else if (recentlyFocusedItems[1] && recentlyFocusedItems[1].assignedNodes && recentlyFocusedItems[1].assignedNodes().length) {
+			this._recentlyFocused.focusItem(recentlyFocusedItems[1].assignedNodes()[0]);
+		}
+	}
+
+	_resetButtonDisabled() {
+		// check for current and saved state and disable Reset button if there is no difference
+		this._resetButton.disabled = JSON.stringify(this._settings) === JSON.stringify(this._initialSettings);
+	}
+
+	_acceptSettings() {
```

Comment:
Maybe this function can share the name with the event, as it is with 'cancelSettings', for example 'submitSettings' or 'confirmSettings'

Reply:
done